_posts/2020-06-13-IrishGen-RDF-Linked-Data-Semantic-Web.md

```diff
@@ -0,0 +1,354 @@
+ecosystem of terms.  The choices made do not come without cost and
+without their own challenges.  This section will explore some of them
+that IrishGen has encountered over the years and give some solutions
+to those which may be of interest to readers.
+
+Human error is the bane for may digital curation projects.  IrishGen
+is no less effected.  While typos are generally an annoyance, in
+IrishGen and RDF generally, they can cause entire systems to go awry.
+A case in point, a vast number of the entries in IrishGen have the
+form below:
+
+```turtle
+<#Flaithbertach>
+    a foaf:Person;
+    irishRel:nomName "Flaithbertach";
+    rel:childOf <#Crunmael>.
+```
+
+This states that Flaithbertach is a person who has the nominative name
+Flaithbertach and is the child of someone called Crunmael.  While
+automated generation of this entry can help (as discussed in [Human
+Curation and Digital Datasets: A Problem in Multiple
+Parts]({{site.baseurl}}/2020/06/07/Human-Curation-and-Digital-Datasets.html)),
+this still requires human intervention and thus the possibility of
+error.  If an error is introduced of the form:
+
+```
+<#Flaithbertach>
+    a foaf:Person;
+    irishRel:nomName "Flaithbertach";
+    rel:childOF <#Crunmael>.
+
+```
+
+Notice the difference between `childOf` and `childOF`.  This is subtle
+and easily missed even during a full audit.  The effect of this change
+is that the predicate is assumed to exist but it has no effect.
+Essentially, this means that in the presence of a Triplestore that can
+apply reasoning to a triple, no reasoning will apply.  The implication
+of this is that now Flaithbertach is not the child of Crunmael and
+when searched for under its proper form, `childOf`, it will not
+appear.  The system is broken and if this is in a complex query,
+incorrect results will be returned to the user.  The old adage
+"Garbage In, Garbage Out" applies but for a user who is not facile
+with computers and is not ready to expect this in a human curated
+database, it can be very frustrating and confidence draining.
+
+There is not much that can be done for this kind of problem overall.
+More careful curators and automation can decrease the occurance as
+discussed in "Human Curation and Digital Datasets" but this will never
+fully elimiate the possibility.
+
+Another form of error can cause real problems in a system.  Take the
+below more fleshed out version:
+
+```turtle
+@base <http://example.com/LL/dal_corpri_arad.trig> .
+@prefix foaf: <http://xmlns.com/foaf/0.1/> .
+@prefix rel: <http://purl.org/vocab/relationship/> .
+
+<http://example.com/LL> {
+  <#Flaithbertach> a foaf:Person ;
+                   rel:childOf <#Crunmael>.
+}
+```
+
+The above is a full TRiG file that will be parsed successfully by a
+system that can parse the TRiG file format.  The file above translated
+into English states: "In LL, Flaithbertach is a person and is the
+child of Crunmael".  As the astute reader will have noticed, Crunmael
```

Comment:
See comments above about whether they can properly be said to be "called" Flaithbertach and Crunmael. I don't know how relevant they really are.